mockgen/mockgen.go

```diff
@@ -225,6 +225,12 @@ func (g *generator) Generate(pkg *model.Package, pkgName string) error {
 		localNames[pkgName] = true
 	}
 
+	// Write a package-level documentation comment only if emitting a new
+	// package outright, which would not clobber the original's
+	// documentation.
+	if inPkg, outPkg := pkg.Name, pkgName; inPkg != outPkg {
```

Comment:
Hi, @balshetzer, thank you for the immediate attention!

I imagined it wouldn't be 100 percent comprehensive.  I would be gravely concerned from a developer experience perspective if GoMock overwrote *without trying* to check whether it would overwrite.

Looking at our monorepo, I estimate (by informal sampling) …

1. we explicitly set an output package name 50 percent of time

2. we overwhelmingly set the output package name to a divergent name from the original with the nomenclature for these being all across the board — ca. 95 percent of the time

Are there any cheap ways we could further enhance this without wandering around the filesystem aimlessly?  We use [Bazel](https://bazel.build) extensively, so a free filesystem walk is out of the question, since we must accurately declare build inputs and their dependencies in advance.

Alternatively, this check could be sufficient given pt. 2.  From a perspective of good Go design, I cannot imagine too many folks bundling GoMock output with their production code's exported types in the same package.  The reason is cluttering the minimum viable set of types required to use a package.  Perhaps if someone attempts to to emit to a pre-existing package of production export typed — considering the best practice I highlighted — then they should be able to accept the risks of our good faith attempt.

IIUC, godoc choses the package's documentation from the first file it visits in the directory which contains a package-level documentation comment.